executor/executor_write.go

```diff
@@ -234,3 +236,101 @@ func (e *UpdateExec) Fields() []*ast.ResultField {
 func (e *UpdateExec) Close() error {
 	return e.SelectExec.Close()
 }
+
+// DeleteExec represents a delete executor.
+// See: https://dev.mysql.com/doc/refman/5.7/en/delete.html
+type DeleteExec struct {
+	SelectExec Executor
+
+	ctx        context.Context
+	Tables     []*ast.TableName
+	MultiTable bool
+}
+
+// Next implements Execution Next interface.
+func (e *DeleteExec) Next() (*Row, error) {
```

Comment:
At least, add a `finished` flag to prevent it from executing multiple times if `Next` is called multiple times.